compiler/src/dotty/tools/dotc/core/Definitions.scala

```diff
@@ -1111,10 +1111,49 @@ class Definitions {
    *  is read from a classfile.
    */
   def patchStdLibClass(denot: ClassDenotation)(using Context): Unit =
-
     def patch2(denot: ClassDenotation, patchCls: Symbol): Unit =
       val scope = denot.info.decls.openForMutations
+
       def recurse(patch: Symbol) = patch.is(Module) && scope.lookup(patch.name).exists
+
+      def makeClassSymbol(patch: Symbol, parents: List[Type], selfInfo: TypeOrSymbol) =
+        newClassSymbol(
```

Comment:
Why can't we use `patch.copy(...)` here? Would that not be shorter?

Reply:
The reason is that `info` and `symbol` depend on each other, and `Symbol.copy` does not support that.